pre_commit_hooks/trailing_whitespace_fixer.py

```diff
@@ -3,17 +3,33 @@
 import argparse
 import fileinput
 import sys
+import os
 
 from pre_commit_hooks.util import cmd_output
 
 
-def _fix_file(filename):
+def _fix_file(filename, markdown=False):
     for line in fileinput.input([filename], inplace=True):
+        # preserve trailing two-space for non-blank lines in markdown files
+        if markdown and (not line.isspace()) and (line.endswith("  \n")):
+            line = line.rstrip(' \n')
+            # only preserve if there are no trailing tabs or unusual whitespace
+            if not line[-1].isspace():
+                print(line + "  ")
+                continue
```

Comment:
I used the `continue` to avoid repeating code. Since there are nested `if` statements (with a required rstrip and assignment between the two conditions, making it impractical to combine the two if statements using `and`), the alternatives are actually
```
loop:
    if ...:
        ...0
        if ...:
            ...1
            continue
    ...2
```
or
```
loop:
    if ...:
        ...0
        if ...:
            ...1
        else:
            ...2
    else:
        ...2
```
In this case, the repeated code is only `print(line)` so not such a big deal, let me know if you prefer that to the current use of `continue` and I'll change it.